automaxprocs.go

```diff
@@ -18,37 +18,86 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// +build linux
-
 package automaxprocs
 
 import (
-	"log"
 	"os"
 	"runtime"
 
 	iruntime "go.uber.org/automaxprocs/internal/runtime"
 )
 
-func init() {
+const _maxProcsKey = "GOMAXPROCS"
+
+func currentMaxProcs() int {
+	return runtime.GOMAXPROCS(0)
+}
+
+type config struct {
+	printf func(string, ...interface{})
+	procs  func(int) (int, iruntime.CPUQuotaStatus, error)
+}
+
+func (c *config) log(fmt string, args ...interface{}) {
+	if c.printf != nil {
+		c.printf(fmt, args...)
+	}
+}
+
+// An Option alters the behavior of Set.
+type Option interface {
+	apply(*config)
+}
+
+// Logger uses the supplied printf implementation for log output. By default,
+// Set doesn't log anything.
```

Comment:
I split the difference and kept `maxprocs` silent by default, but had `automaxprocs` send output to `log.Printf`.